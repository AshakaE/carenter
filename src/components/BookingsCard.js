import React from 'react';
import PropTypes from 'prop-types';
import { urlDeleteBooking } from '../assets/getAuth';
import BookingsForm from './forms/BookingsForm';

const BookingsCard = (props) => {
  const [isMember, setIsMember] = React.useState(true);
  const [update, setUpdate] = React.useState(false);
  const [msg, setMsg] = React.useState('');
  const {
    item: {
      id, name, carModel, carName, imageUrl, userName, price, carId,
    },
  } = props;

  const handleDelete = async () => {
    const response = await urlDeleteBooking(id);
    const msg = response.message;
    setMsg(msg);
  };

  const toggleHidden = () => {
    setIsMember(!isMember);
    setUpdate(!update);
  };

  return (
    <>
      <div>{name}</div>
      <div>{carModel}</div>
      <div>
        <img src={imageUrl} alt={carName} />
      </div>
      <div>{carName}</div>
      <div>{userName}</div>
      <div>{price}</div>
      <button type="button" onClick={handleDelete}>
        Delete booking
      </button>
      <button type="button" onClick={toggleHidden}> Update booking</button>
      {!isMember && <BookingsForm carId={id} id={carId} update={update} alert={alert} />}
      {msg && (
        <div>{msg}</div>
      )}
    </>
  );
};

BookingsCard.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number,
    carId: PropTypes.number,
    price: PropTypes.number,
    name: PropTypes.string,
    carName: PropTypes.string,
    carModel: PropTypes.string,
    userName: PropTypes.string,
    imageUrl: PropTypes.string,
  }),
};

BookingsCard.defaultProps = {
  item: [],
};

export default BookingsCard;
