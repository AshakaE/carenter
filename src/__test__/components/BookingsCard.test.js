import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BookingsCard from '../../components/BookingsCard';

describe('BookingsCard', () => {
  it('renders correctly', () => {
    const id = 1;
    const name = 'test';
    const carModel = 'test';
    const carName = 'test';
    const userName = 'test';
    const price = 1000;
    const carId = 1;
    const date = 'test';
    const tree = renderer
      .create(
        <BrowserRouter>
          <BookingsCard
            item={{
              name,
              carModel,
              carName,
              userName,
              price,
              carId,
              id,
              date,
            }}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
