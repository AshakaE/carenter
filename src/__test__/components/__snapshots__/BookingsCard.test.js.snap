// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingsCard renders correctly 1`] = `
Array [
  <div
    className="gridBox"
  >
    <div>
      test
    </div>
    <div>
      test
    </div>
    <div>
      test
    </div>
    <div>
      test
    </div>
    <div>
      1000
    </div>
    <div>
      test
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
         
        Update booking
      </button>
      <button
        className="btnDel"
        onClick={[Function]}
        type="button"
      >
        Delete booking
      </button>
    </div>
  </div>,
  <div
    className="form"
  >
    <div>
      
    </div>
  </div>,
]
`;
